### PR TITLE
Patient

### PR DESCRIPTION
**Pull request by OlenkaSheremet**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -35,6 +35,7 @@
         </dependency>
```
Comment (rublin) [inline pom.xml]: Please remove empty extra lines. It is not your fault, but I'd like to merge clean code

File: src/main/java/com/mydoctor/mydoctorbot/model/Patient.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import javax.persistence.Id;
+import java.time.LocalDateTime;
+
+@Entity
+@Getter
+@Setter
+@ToString
+public class Patient {
+    @Id
+    @GeneratedValue(strategy= GenerationType.IDENTITY)
```
Comment (rublin) [inline src/main/java/com/mydoctor/mydoctorbot/model/Patient.java]: it looks like you forgot to add a space before the equals symbol. It is a good practice to run code format before commit changes (ctrl + alt + shift + L in idea)

File: src/main/java/com/mydoctor/mydoctorbot/model/Patient.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    private Long id;
+    private String firstName;
+    private String lastName;
+    private LocalDateTime dateOfBirth;
+    private String typeOfDoctor;
```
Comment (rublin) [inline src/main/java/com/mydoctor/mydoctorbot/model/Patient.java]: We don't need typeOfDoctor inside the patient. We will store relations between the patient and the doctor in another table.

File: src/main/java/com/mydoctor/mydoctorbot/service/PatientService.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        patient.setLastName(lastName);
+        patient.setTypeOfDoctor(typeOfDoctor);
+
+        Patient saved = patientRepository.save(patient);
+        return saved;
+
+    }
+
+    public List<Patient> getaAll() {
+
```
Comment (rublin) [inline src/main/java/com/mydoctor/mydoctorbot/service/PatientService.java]: Please remove this empty line

File: src/main/resources/db/migration/V2__create_patient_table.sql [hunk 5]
```diff
@@ -0,0 +1,16 @@
+create table patient
+(
+	id bigserial not null,
+	first_name varchar not null,
```
Comment (rublin) [inline src/main/resources/db/migration/V2__create_patient_table.sql:4]: for the doctor, we need the full name and they are required fields. but for the patient, we shouldn't take name and surname as required fields. so, my suggestion is not to mark them as required (not null).

**Review by rublin — CHANGES_REQUESTED** [review]
@OlenkaSheremet Good work, I really proud of you. Please fix these small comments and I'm going to approve the PR

File: pom.xml (hunk 1 above)
Comment (OlenkaSheremet) [reply pom.xml]: done

**Review by rublin — APPROVED** [review]

**Review by aliska2 — APPROVED** [review]
approve

**Pull request merged**